src/lsst/rsp/utils.py

```diff
@@ -73,63 +81,99 @@ def jupyter_proxy_url(port: Optional[int]) -> str:
         full_url = urllib.parse.urljoin(user_url, proxy_url_path)
         return full_url
 
-    bokeh.io.show(obj, notebook_url=jupyter_proxy_url)
+    bokeh.io.show(obj=obj, notebook_url=jupyter_proxy_url())
```

Comment:
You're now calling the function instead of passing the function into bokeh. If you can pass in a static string, I assume you should get rid of the nested def now since it's not serving any purpose?

Reply:
It always should have been a string: https://docs.bokeh.org/en/latest/docs/reference/io.html#bokeh.io.show

Part of getting CI actually working today was that `tox -e typing` found this for me.

The nested function does a fair bit of work to cough up that string, though, so I don't think we want to try to do all of that in the show() params.

Reply:
Oh, no, I see what was happening.  The type hints for Bokeh are wrong.

notebook_url can also be a function that takes one int for the bound server port. If the port is provided, the function needs to generate the full public URL to the bokeh server. If None is passed, the function is to generate the origin URL.